pkg/reconciler/common/images.go

```diff
@@ -41,34 +41,80 @@ var (
 	delimiter             = "/"
 )
 
+// ImageTransformer is an interface for transforming images passed to the ResourceImageTransformer
```

Comment:
Maybe we could upstream this into manifestival?

Reply:
That would work for me. @jcrossley3 what do you think about Manifestival providing some out of the box transformations. The specific `v1alpha1.Registry` transformation would continue to live in this operator code but the `ImageTransformer` would live in `manifestival`. I'm happy to do the port.

Reply:
I like having MF provide ootb transformers, e.g. namespace and ownerref, but is there much value to other non-knative MF users without a default impl?

Reply:
IMO, there would be value to providing Identity and ChangeRepo implementations (and likely more, if we thought about it :). I think needing to swap out a personal repository while testing / use different repositories for different locations/shards/environments would be fairly frequent use cases for anything manipulating a manifest.

Reply:
We could provide a base impl for swapping a regex with another string:

```
type RegexpReplacer struct {
  ToReplace string
  ReplaceWith string
}

var _ ImageTransformer = (*RegexpReplacer)(nil)

func (rr *RegexpReplacer) HandleImage(containerName, parent string, useDefault bool) (string, bool) {
  strings.ReplaceAll(containerName, rr.ToReplace, rr.ReplaceWith)
}

func replaceName(imageTemplate string, name string) string {
	return strings.ReplaceAll(imageTemplate, containerNameVariable, name)
}

func (rit *registryImageTransformer) HandleImagePullSecrets(imagePullSecrets []corev1.LocalObjectReference, log *zap.SugaredLogger) []corev1.LocalObjectReference {
	return imagePullSecrets
}
```